### PR TITLE
Add fig generate completion subcommand

### PR DESCRIPTION
**Pull request by miguelmig**
Add a subcommand to `generate` to create a Fig Spec which can be integrated with Amazon CloudWhisperer CLI/Amazon Q/Fig.

Example usage:
`usage g fig -f mise.usage.kdl --out-file tasks/fig/src/mise.ts`

Can be compiled into javascript using the following command on `tasks/fig` folder: `npx @withfig/autocomplete-tools compile` and used locally by moving the resulting `build/mise.js` into `~/.fig/autocomplete/build` (if the folders don't exist just create them)

### CODE REVIEW

**Review by jdx — CHANGES_REQUESTED** [review]
minor feedback, can you also add a section for this to the docs? https://usage.jdx.dev/cli/completions

File: tasks/fig/generators.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    })
+
+    const subcommands = (await Promise.allSettled(promises)).filter(p => p.status === 'fulfilled').map(p => p.value);
+    
+    return { subcommands: subcommands.flat() } as Fig.Spec
+  }
+}
+
+
+export const pluginGenerator: Fig.Generator = singleCmdNewLineGenerator('mise plugins --core --user')
```
[the remark this answers is not in the document]
Comment (miguelmig) [reply tasks/fig/generators.ts:95]: My bad, this is a part of a different PR for the mise repo with specialized generators. Will remove now.

File: cli/Cargo.toml [hunk 2]
```diff
@@ -42,6 +42,7 @@ tera = "1"
 thiserror = "1"
 usage-lib = { workspace = true, features = ["clap", "docs"] }
 xx = "1"
+serde_json = "1.0"
```
[the remark this answers is not in the document]
Comment (miguelmig) [reply cli/Cargo.toml]: Addressed in 84cf3a7

File: tasks/fig/generators.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (miguelmig) [reply tasks/fig/generators.ts]: Addressed in 84cf3a7

File: cli/src/cli/generate/mod.rs [hunk 3]
```diff
@@ -17,13 +18,15 @@ pub struct Generate {
 pub enum Command {
     Completion(completion::Completion),
     Markdown(markdown::Markdown),
+    Fig(fig::Fig),
```
[the remark this answers is not in the document]
Comment (miguelmig) [reply cli/src/cli/generate/mod.rs:20]: Addressed in 84cf3a7

**Pull request merged**